examples/example_analysis_workflows.py

```diff
@@ -0,0 +1,186 @@
+
+    observations = get_observations()
+    target = Target(pos='dummy', on_region='dummy')
+    config = dict(method='dummy', apply_psf_correction=True)
+
+    # User can run this explicitly, can be done via spectrum_extraction.estimate_bg()
+    # bg_estimator = BackgroundEstimator(observations, target, **config)
+    # bg_estimator.run() # computes reflected regions and filters off lists
+    #
+    # # If you want, the `BackgroundEstimate` is serialisable
+    # bg_estimator.background_estimate.write(dir='temp')
+    # bg_estimate = BackgroundEstimate.read(dir='temp')
+
+    spectrum_extraction = SpectrumExtraction(bg_estimate, observations, target, **config)
+    spectrum_extraction.run()
+
+    # spectrum_extraction.make_reflected_regions()
+    # spectrum_extraction.bg_estimate
+    # spectrum_extraction.estimate_background()
+    # stats_table = spectrum_extraction.make_stats_table()
+
+    # If you want, the `SpectrumObservationList` is serialisable
+    spectrum_extraction.spectrum_observation_list.write(dir='temp')
+    spectrum_obs_list = SpectrumObservationList.read(dir='temp')
+
+    # Compute new stacked spectra with a given grouping
+    group_id = 'left up to the user for now'  # Python list of group_id for each obs_id
+    spectrum_obs_list = spectrum_obs_list.apply_stacking(group_id=group_id)
+
+    model = PowerLaw()
+    spectrum_fit = SpectrumFit(spectrum_obs_list, model)
+    spectrum_fit.run()
+
+    # If you want, the `SpectrumResult` is serialisable
+    spectrum_fit.spectrum_result.write(dir='temp')
+    spectrum_fit_result = SpectrumFitResult.read(dir='temp')
+
+
+    # analysis = GtAnalysis(observations, config)
+    # analysis.setup()
+    # analysis.fit()
+
+
+def run_image_analysis():
+    """Run 2-dim image analysis.
+
+    """
+    # from gammapy.scripts import MosaicImage, ObsImage
+    # ObsImage is equivalent to SpectrumObservation
+    # MosaicImage is equivalent to SpectrumObservationList and SpectrumObservation
+    from gammapy.image import SkyMap
+    from gammapy.image import MapMaker, MapFitter, SkyMaps
+
+    observations = get_observations()
+
+    ref_map = SkyMap.empty('dummy')
+
+    config = dict(method='ring')
+    maker = MapMaker(observations, ref_map, **config)
+    maker.run()
+
+    maker.images
+
+    stats_table = maker.make_stats_table()
+
+
+def run_cube_analysis():
+    """Run 3-dim cube analysis.
+
+    """
+    observations = get_observations()
+
+
+def analyse_target_list():
+    """Example how to process multiple targets.
+
+    This example runs two analyses for each target:
+    - 1-dim spectral
+    - 2-dim image analy
+    """
+    from gammapy.spectrum import SpectrumExtraction, SpectrumFit
+    from gammapy.image import ImageMaker, ImageFitter
+    from gammapy.scripts import TargetList
+
+    observations = get_observations()
+    target_list = TargetList()
+
+    for target in target_list:
+        config = dict()
+        analysis = SomeAnalysis(**config)
```

Comment:
Here you need to define a way to translate a config dict (probably read from YAML) into a dict with gammapy/astropy classes as values